src/vivarium_census_prl_synth_pop/components/population.py

```diff
@@ -594,11 +603,119 @@ def get_household_structure(pop: pd.DataFrame) -> pd.DataFrame:
+        ].index
+        # Potential partner index groups
+        partners_idx = mothers_households.loc[
+            mothers_households["member_relation_to_household_head"].isin(partners)
+        ].index
+        reference_person_idx = mothers_households.loc[
+            mothers_households["member_relation_to_household_head"] == "Reference person"
+        ].index
+
+        # Assign second guardian to random partner of mothers
+        partner_ids = mothers_households.loc[mother_rp_idx.intersection(partners_idx)]
+        if len(partner_ids) > 0:
+            partner_ids = self.choose_random_guardian(partner_ids, "mother_id")
+            new_births.loc[
+                new_births["parent_id"].isin(partner_ids.index), "guardian_2"
+            ] = new_births["parent_id"].map(partner_ids)
+
+        reference_person_ids = mothers_households.loc[
+            mother_partner_idx.intersection(reference_person_idx)
+        ]
+        if len(reference_person_ids) > 0:
+            reference_person_ids = self.choose_random_guardian(
+                reference_person_ids, "mother_id"
+            )
+            new_births.loc[
+                new_births["parent_id"].isin(reference_person_ids.index), "guardian_2"
+            ] = new_births["parent_id"].map(reference_person_ids)
+
+        return new_births
+
+    @staticmethod
+    def get_mothers_household_structure(
+        mothers_idx: pd.Series, households: pd.DataFrame
+    ) -> pd.DataFrame:
+        """
+
+        Parameters
+        ----------
+        mothers_idx: Series of index values corresponding to mothers who gave birth this time step
+        households: 2 column dataframe of state table with containing "household_id" and "relation_to_household_head"
+          columns.
+
+        Returns
+        -------
+        Dataframe with 2 index levels "mother_id" and "person_id" and 2 columns "mother_relation_to_household_head" and
+          "member_relation_to_household_head".  This is the same schema we used with creating the data structure for
+          child households.
+        """
+        # todo: Future improvement would be to refactor the two data wrangling functions.
+
+        mothers = (
+            households.loc[mothers_idx]
+            .reset_index()
+            .rename(columns={"index": "mother_id"})
+            .set_index(["household_id", "mother_id"])
+        )
+        household_info = (
+            households.reset_index()
+            .rename(columns={"index": "person_id"})
+            .set_index(["household_id", "person_id"])
+        )
+
+        mother_households = mothers.merge(household_info, left_index=True, right_index=True)
+        mother_households.rename(
+            columns={
+                "relation_to_household_head_x": "mother_relation_to_household_head",
+                "relation_to_household_head_y": "member_relation_to_household_head",
+            },
+            inplace=True,
```

Comment:
nit: you `inplace=True` here but then overwrite in the next `.droplevel` step. Consistency is nice

Reply:
+1